homeassistant/util/unit_system.py

```diff
@@ -120,6 +121,7 @@ def __init__(
         self.pressure_unit = pressure
         self.volume_unit = volume
         self.wind_speed_unit = wind_speed
+        self.length_conversions = length_conversions
```

Comment:
I know this it a proof of concept, but I think here we should use a private field:
```suggestion
        self._length_conversions = length_conversions
```

Then expose it as `def get_converted_unit(self, device_class, original_unit) -> str | None:`